int32 id
float32 speed